src/text/parsers/containers.rs

```diff
@@ -0,0 +1,109 @@
+use crate::text::TextStreamItem;
+use nom::branch::alt;
+use nom::bytes::streaming::tag;
+use nom::combinator::{map, recognize};
+use nom::IResult;
```

Comment:
This file defines `nom` parsing rules to detect the beginnings and endings of the various container types.